clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerPartitionAssignor.java

```diff
@@ -178,6 +178,21 @@ public GroupAssignment(Map<String, Assignment> assignments) {
         }
     }
 
+    /**
+     * {@link RebalanceProtocol#EAGER} rebalance protocol will let a consumer to always revoke all its owned partitions
+     * before participating the rebalance event, and therefore allows a complete reshuffling of the assignment.
+     *
+     * {@link RebalanceProtocol#COOPERATIVE} rebalance protocol will let a consumer to retain its currently assigned
```

Comment:
This sentence really goes on. How about 

`{@link RebalanceProtocol#COOPERATIVE} rebalance protocol allows consumers to retain their currently owned partitions before entering a rebalance, and only revoke those partitions which will actually be migrated to other consumers based on the new assignment. The actual assignment will be adjusted so that partitions which are to be revoked will not be reassigned to their new owners during the same rebalance, but must wait for another rebalance to be assigned.`